docs: update getting started for otel-swift and otel-swift-core

williazz opened the pull request:
Resolves https://github.com/open-telemetry/opentelemetry-swift/issues/908

williazz committed:
docs: update getting started for otel-swift and otel-swift-core

ArielDemarco reviewed (commented):
PR Looks fine. I'd also include the changes related to this when talking about cocoapdos (fwiw, is also outdated as it mentions we only have 2 pods in this repo while we have 3)

williazz committed:
fix: resovle cocoapods

williazz committed:
docs: bump versions in Getting Started to latest releases

Update the Package.swift snippet to reference the latest published
versions: opentelemetry-swift 2.3.0 and opentelemetry-swift-core 2.4.1.
The two packages are released independently, so their versions can
diverge.
